### PR TITLE
Replacing lodash invoke

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Internal cleanups (small changes):
1) Replacing lodash invoke with optional chaining (fix types as a result).
2) Remove Date as a type.
3) Fix lodash imports on spec files.
4) 'input.js' - renaming 'velocityTracker'
5) 'test.js'  - renaming 'testUtils'
6) Removing unused parameter 'doNotTriggerListeners' from updateMonth() on Calendar and CalendarList

### CODE REVIEW

File: src/agenda/reservation-list/index.tsx [hunk 1]
```diff
@@ -246,7 +245,7 @@ class ReservationList extends Component<ReservationListProps, ReservationsListSt
     const dateIsSame = sameDate(day, this.selectedDay);
     if (!dateIsSame && this.scrollOver) {
       this.selectedDay = day.clone();
-      invoke(this.props, 'onDayChange', day.clone());
+      this.props.onDayChange?.(new Date(day.getUTCMilliseconds()));
```
Comment (lidord-wix) [inline src/agenda/reservation-list/index.tsx:248]: why changing from `day.clone()` to `new Date(day.getUTCMilliseconds())`?

**Review by lidord-wix — COMMENTED** [review]
I left a small question just out of curiosity.. but looks good overall

File: src/agenda/reservation-list/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx:248]: type issue, but seams the type on the prop is wrong so I'll change it again

Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx:248]: Changed the type on the onDayChange prop

**Comment by Inbal-Tish** [comment]
@lidord-wix Don't merge just yet. There is another PR with a critical issue there that I need to merge first and then I'll solve the conflicts here

**Comment by Inbal-Tish** [comment]
@lidord-wix Since we had a ts bug that resulted in an actual return value bug, I added a section to this PR where I remove all Date types and related conversions. We should never have 'Date' as a type here...

**Review by lidord-wix — APPROVED** [review]

**Comment by lidord-wix** [comment]
Approved. 
Merge if you done with the other PR :)

**Comment by Inbal-Tish** [comment]
> Approved. Merge if you are done with the other PR :)

Yes, the other pr was already merged to this one. Thanks

**Pull request merged**